### PR TITLE
Add syscon functions

### PR DESCRIPTION
**Pull request by egelja**
Try it out:
1. Run the kernel
2. Press `^`
3. Press `&`

### CODE REVIEW

File: include/kernel/of/syscon.h
Comment (kjhayes) [inline include/kernel/of/syscon.h:1]: This doesn't belong in the "kernel/of" directory, it's a really primitive driver and "of" is for OpenFirmware stuff

File: docs/arch.md
Comment (kjhayes) [inline docs/arch.md:1]: This file is gonna get loooooooooooong but I'm glad to see we're planning ahead

Comment (kjhayes) [reply docs/arch.md:1]: Also, theoretically it should be drivers which try to call `register_early_putchar` not necessarily the architecture

**Review by kjhayes — CHANGES_REQUESTED** [review]
I guess it's okay for now but the main thing is `syscon` definitely doesn't belong in "kernel/of", it might be time to make "kernel/drivers" or "kernel/dev/" or just "drivers"

**Comment by kjhayes** [comment]
Overall really cool, I'm glad the FDT stuff wasn't too awful to use

File: docs/arch.md
Comment (egelja) [reply docs/arch.md:1]: Ya I guess it would be better to says "initializes basic drivers" right?

File: drivers/syscon/syscon.c
Comment (kjhayes) [inline drivers/syscon/syscon.c:1]: I'm gonna be pushing a commit to `fdt_mem` soon which should make this a lot simpler so one of us should come back and use the new functions in `fdt.h` (For reference the issue currently is that it's not taking into account `#address-cells` or `#size-cells` when reading `reg`, dw about it though)

Comment (kjhayes) [reply drivers/syscon/syscon.c:1]: But looks good

**Review by kjhayes — APPROVED** [review]

**Pull request merged**